### PR TITLE
SONARIAC-572 Adapt HereDocForm to the new grammar

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/HereDocumentImpl.java [hunk 1]
```diff
@@ -45,13 +46,12 @@ public Kind getKind() {
 
   @Override
   public List<SyntaxToken> literals() {
-    return List.of(content);
+    throw new UnsupportedOperationException("TODO SONARIAC-579 Remove LiteralList.literals()");
   }
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/HereDocumentImpl.java:50]: By throwing an exception in this method, it is stopping rules currently using this method from running, which prevent us from seeing what is the impact of this PR on the ruling execution.
If possible, I would recommend to have a temporarily replacement of this method content like we did for ShellForm and ExecForm, you can very probably just copy/paste the same logic there.

**Review by rudy-regazzoni-sonarsource — CHANGES_REQUESTED** [review]
Great work ! Please see my comments, few elements to adapt.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/HereDocumentImpl.java (hunk 1 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/HereDocumentImpl.java]: 👍 Perfectly fine, we didn't cover also as this is temp code that is soon to be removed.

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM !

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/utils/ArgumentUtils.java [hunk 2]
```diff
@@ -76,6 +79,15 @@ private static String resolveExpression(Expression expression) {
...
   }
 
+  @Nullable
+  public static SyntaxToken argumentToSyntaxToken(Argument argument) {
+    String value = ArgumentUtils.resolve(argument).value();
+    if (value != null) {
+      return new SyntaxTokenImpl(value, argument.textRange(), Collections.emptyList());
+    }
+    return null;
+  }
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/utils/ArgumentUtils.java:89]: Please add a comment. We should remove this helper after we drop all `literals()` methods.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/HereDocumentImpl.java [hunk 3]
```diff
@@ -45,13 +46,16 @@ public Kind getKind() {
 
   @Override
   public List<SyntaxToken> literals() {
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/HereDocumentImpl.java:48]: A comment is missing to drop the method.

**Review by nils-werner-sonarsource — COMMENTED** [review]
Please add a few comments to track the method which we should remove.

**Pull request merged**